newIDE/app/src/SceneEditor/index.js

```diff
@@ -1136,6 +1178,73 @@ export default class SceneEditor extends React.Component<Props, State> {
     ];
   };
 
+  getContextMenuLayoutItems = (i18n: I18nType) => [
+    {
+      label: i18n._(t`Open scene events`),
+      click: () => this.props.onOpenEvents(this.props.layout.getName()),
+    },
+    {
+      label: i18n._(t`Open scene properties`),
+      click: () => this.openSceneProperties(true),
+    },
+  ];
+
+  getContextMenuInstancesWiseItems = (i18n: I18nType) => [
+    {
+      label: i18n._(t`Copy`),
+      click: () => this.copySelection(),
+      enabled: this.instancesSelection.hasSelectedInstances(),
```

Comment:
You can probably save `const hasSelectedInstances = this.instancesSelection.hasSelectedInstances()` at the top of this function to re-use it ad not call it for every item

Reply:
Thanks, I did not realize it was called for each item 😅